File: app/src/main/java/com/marked/pixsee/features/chat/custom/MessageHolder.java
package com.marked.pixsee.features.chat.custom;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.marked.pixsee.R;
import com.marked.pixsee.features.chat.data.Message;
import com.marked.pixsee.features.chat.data.MessageConstants;


/**
 * Created by Tudor Pop on 04-Dec-15.
 */
public class MessageHolder extends RecyclerView.ViewHolder implements Bindable{
	private TextView mMessageTextView;

	public MessageHolder(View itemView) {
		super(itemView);
		mMessageTextView = (TextView) itemView.findViewById(R.id.messageTextView);

	}

	@Override
	public void bindMessage(final Message message, final ChatAdapter.ChatInteraction chatInteraction) {
		itemView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				chatInteraction.chatClicked(v, message, getAdapterPosition());
			}
		});
		mMessageTextView.setText(message.getData().get(MessageConstants.DATA_BODY));
	}

}

File: app/src/main/java/com/marked/vifo/ui/adapter/viewholders/MessageHolder.kt
package com.marked.vifo.ui.adapter.viewholders;

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import com.marked.vifo.R
import com.marked.vifo.extra.MessageConstants
import com.marked.vifo.model.message.Message

/**
 * Created by Tudor Pop on 04-Dec-15.
 */
class MessageHolder(itemView: View,context: Context) : RecyclerView.ViewHolder(itemView) {
    private val mMessageTextView = itemView.findViewById(R.id.messageTextView) as TextView
    private val mContext = context

	fun bindMessage(message: Message) {
		mMessageTextView.text = message.data.get(MessageConstants.DATA_BODY)
	}
}
